Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp

```diff
@@ -765,27 +767,17 @@ class MultiEigenStepperLoop
   /// @param release [in] boolean to trigger step size release
   template <typename object_intersection_t>
   void updateStepSize(State& state, const object_intersection_t& oIntersection,
-                      bool release = true) const {
-    const Surface& surface = *oIntersection.representation;
+                      Direction direction, bool release = true) const {
```

Comment:
this is rather ugly but I don't see a better solution atm

ideas @paulgessinger @benjaminhuth ?

Reply:
Hmm another possibility would be to make the direction a member of the `MultiEigenStepperLoop::State` and force its initializationi at the beginning of the propagation.
That would keep the interface cleaner...

Reply:
the problem I see with making the stepper aware of the propagation direction is that we need to keep it aligned. some code might flip it in the propagator options on the go but forget about the stepper state property

Reply:
yeah makes sense. then I think this is the cleanest version.